entity-stats/src/main/java/eu/europeana/entity/stats/service/UsageStatsService.java

```diff
@@ -0,0 +1,151 @@
+     *
+     * @param metric
+     */
+    public void getStatsForLang(EntityMetric metric) throws UsageStatsException {
+       // for total entities : query=*&profile=facets&facet=type&pageSize=0
+       EntityStats entityTotal = getFacetsResults(buildSearchQuery(UsageStatsFields.QUERY_ALL, UsageStatsFields.FACET));
+       if (entityTotal == null) {
+           throw new UsageStatsException(" There are no entities present for facet=type. No stats will be generated");
+       }
+        List<EntitiesPerLanguage> entitiesPerLanguageList = new ArrayList<>();
+        for(String lang : languages) {
+            EntitiesPerLanguage entitiesPerLanguage = getEntityPerLangValues(lang, entityTotal);
+            if(entitiesPerLanguage != null) {
+                entitiesPerLanguageList.add(entitiesPerLanguage);
+            } else {
+                LOG.error("No stats found for lang {} in entity api", lang);
+            }
+        }
+        metric.setEntities(entitiesPerLanguageList);
+    }
+
+    /**
+     * Retrieves Entity stats for the given lang
+     * query=skos_prefLabel.<lang>:*&profile=facets&facet=type&pageSize=0
+     *
+     * @param lang
+     * @param entityTotal
+     * @return
+     * @throws UsageStatsException
+     */
+    private EntitiesPerLanguage getEntityPerLangValues(String lang, EntityStats entityTotal) throws UsageStatsException {
+        EntitiesPerLanguage entityPerLanguage = new EntitiesPerLanguage();
+        StringBuilder query = new StringBuilder(UsageStatsFields.QUERY_SKOS_PREF_LABEL);
+        query.append(".").append(lang).append(":*");
+        // get facet results
+        EntityStats entityStatsForLang = getFacetsResults(buildSearchQuery(query.toString(), UsageStatsFields.FACET));
+        if (entityStatsForLang != null) {
+            // calculate percentage
+            calculatePercentageValues(entityStatsForLang, entityTotal, entityPerLanguage);
+            entityPerLanguage.setLang(lang);
+            return entityPerLanguage;
+        }
+        return null;
+    }
+
+    private static Query buildSearchQuery(String queryString, String facet) {
+        EntityQueryBuilder queryBuilder = new EntityQueryBuilder();
+        return queryBuilder.buildSearchQuery(queryString, null, queryBuilder.toArray(facet), null, 0, 0,
+                SearchProfiles.facets, null);
+    }
+
+    /**
+     * get the value count from the facets results from Solr
+     *
+     * @param searchQuery
+     * @return
+     */
+    private EntityStats getFacetsResults(Query searchQuery) {
+        List<FacetFieldView> facets = solrEntityService.search(searchQuery, null,null, null ).getFacetFields();
+        if (!facets.isEmpty()) {
+            EntityStats entityStats = new EntityStats();
+            // fetch the first facet result view
+            Map<String, Long> map = facets.get(0).getValueCountMap();
+            for (Map.Entry<String, Long> entry : map.entrySet()) {
+                if(entry.getKey().equals("Agent")) {
```

Comment:
we should use EntityTypes.<type>.getInternalTyp() instead for hardcoded values

Reply:
done